tc/core/polyhedral/cuda/mapped_scop.cc

```diff
@@ -107,42 +107,47 @@ const CudaDim& mappingSize<mapping::ThreadId>(const MappedScop* mscop) {
   return mscop->numThreads;
 }
 
-// Map "pos"-th schedule dimension of the band node identified by "tree" to a
-// _new_ parameter identified by "id" and limited by 0 <= id < extent.  The
-// parameter must not be present in the space of partial schedule of "tree" and
-// extent must be non-zero.  The mapping corresponds to inserting a filter
-// node with condition 'dim % extent = id' where dim is "pos"-th
-// schedule dimension.
+// Map the affine functions in "list" to the _new_ parameters
+// identified by id[i] and limited by 0 <= id[i] < extent[i].  The
+// parameters must not be present in the space of partial schedules in "list"
+// and extents must be non-zero.  The mapping corresponds to inserting a filter
+// node with condition 'list % extent = ids'.
+// The mapping is inserted above "tree".
 //
-// Returns a pointer to the updated band (below the inserted filter)
+// Returns a pointer to the updated node (below the inserted filter)
 // for call chaining purposes.
 template <typename MappingIdType>
-detail::ScheduleTree* mapToParameterWithExtent(
+detail::ScheduleTree* mapToParametersWithExtent(
     detail::ScheduleTree* root,
     detail::ScheduleTree* tree,
-    size_t pos,
-    MappingIdType id,
-    size_t extent) {
-  auto band = tree->elemAs<detail::ScheduleTreeElemBand>();
-  CHECK(band) << "expected a band, got " << *tree;
-  CHECK_GE(pos, 0u) << "dimension underflow";
-  CHECK_LT(pos, band->nMember()) << "dimension overflow";
-  CHECK_NE(extent, 0u) << "NYI: mapping to 0";
+    isl::union_pw_aff_list list,
+    const std::vector<MappingIdType>& ids,
+    const CudaDimView& extent) {
+  CHECK_EQ(ids.size(), list.n());
+  CHECK_LE(list.n(), extent.size()) << "dimension overflow";
```

Comment:
I would add a function-level comment mentioning this assumption: if "extent" has more values than "ids", only the first ids.size() values of "extent" will be used.